packages/redux/src/recommendations/types/state.types.ts

```diff
@@ -0,0 +1,14 @@
+import type { CombinedState } from 'redux';
+import type { Error } from '@farfetch/blackout-client/types';
+import type { ProductRecommendation } from '@farfetch/blackout-client/recommendation/types';
+
+export type State = CombinedState<{
+  error: Record<string, Error | null | undefined>;
+  isLoading: Record<string, boolean>;
+  result: ProductRecommendationResult;
+}>;
```

Comment:
State can be undefined as well as the recommendations slice is defined in StoreState.types.ts as being optional. Can you change this, please? Make sure then in all selectors that use a value of this type to handle the undefined case as well.